cmd/jaeger/internal/components.go

```diff
@@ -72,6 +73,7 @@ func (b builders) build() (otelcol.Factories, error) {
 		storagecleaner.NewFactory(),
 		remotesampling.NewFactory(),
 		expvar.NewFactory(),
+		remotestorage.NewFactory(),
```

Comment:
nit: group storagecleaner and remotestorage together, with comment "only for e2e testing"